Auth/FirebaseAuthUI/FUIAuth.m

```diff
@@ -237,12 +237,7 @@ - (void)autoUpgradeAccountWithProviderUI:(id<FUIAuthProvider>)providerUI
     if (error) {
       // Check for "credential in use" conflict error and handle appropriately.
       if (error.code == FIRAuthErrorCodeCredentialAlreadyInUse) {
-        FIRAuthCredential *newCredential = credential;
-        // Check for and handle special case for Phone Auth Provider.
-        if (providerUI.providerID == FIRPhoneAuthProviderID) {
-          // Obtain temporary Phone Auth credential.
-          newCredential = error.userInfo[FIRAuthErrorUserInfoUpdatedCredentialKey];
-        }
+        FIRAuthCredential *newCredential = error.userInfo[FIRAuthErrorUserInfoUpdatedCredentialKey];
```

Comment:
This implementation seems to be provider-specific, so we should check for nil here to avoid crashing on line 242 below.

Reply:
Yeah, good catch.